deps/outliner/src/logseq/outliner/core.cljs

```diff
@@ -232,6 +232,40 @@
          (map (fn [tag]
                 [:db/retract (:db/id block) :block/tags (:db/id tag)])))))
 
+;; TODO: Validate tagless pages and other block types
+(defn- validate-unique-by-name-tag-and-block-type
+  [db new-title {:block/keys [tags] :as entity}]
+  ;; (prn :entity entity)
+  (cond
+    (ldb/page? entity)
+    (when-let [res (seq (d/q '[:find [(pull ?b [* {:block/tags [:block/title]}])]
```

Comment:
there's no need to `pull ?b` here, only used in the exception-branch(`throw ex-info`). 
we should delay this `pull`.

Reply:
Agree, I'll leave this to @logseq-cldwalker.